src/Databases/PostgreSQL/fetchPostgreSQLTableStructure.cpp

```diff
@@ -301,18 +339,38 @@ PostgreSQLTableStructure fetchPostgreSQLTableStructure(
         ? " AND relnamespace = (SELECT oid FROM pg_namespace WHERE nspname = 'public')"
         : fmt::format(" AND relnamespace = (SELECT oid FROM pg_namespace WHERE nspname = {})", quoteString(postgres_schema));
 
-    std::string query = fmt::format(
-           "SELECT attname AS name, " /// column name
-           "format_type(atttypid, atttypmod) AS type, " /// data type
-           "attnotnull AS not_null, " /// is nullable
-           "attndims AS dims, " /// array dimensions
-           "atttypid as type_id, "
-           "atttypmod as type_modifier, "
-           "attgenerated as generated " /// if column has GENERATED
-           "FROM pg_attribute "
-           "WHERE attrelid = (SELECT oid FROM pg_class WHERE {}) "
-           "AND NOT attisdropped AND attnum > 0 "
-           "ORDER BY attnum ASC", where);
+    int version = getVersion(tx);
+
+    std::string query;
+    if (version >= 120000)
```

Comment:
so the difference is only in `attgenerated` ? if so, please add a comment about it that it is not supported in older postgres version and that is why we omit it.

Reply:
better to  support < postgres 12, because some connection is just Postgres protocol compatible, which can not be upgraded to 12

Reply:
better to  support < postgres 12, because some connection is just Postgres protocol compatible, which can not be upgraded to 12

Reply:
better to  support < postgres 12, because some connection is just Postgres protocol compatible, which can not be upgraded to 12